tfe/resource_tfe_organization_token.go

```diff
@@ -68,7 +74,26 @@ func resourceTFEOrganizationTokenCreate(d *schema.ResourceData, meta interface{}
 		log.Printf("[DEBUG] Regenerating existing token for organization: %s", organization)
 	}
 
-	token, err := config.Client.OrganizationTokens.Create(ctx, organization)
+	// Get the token create options.
+	options := tfe.OrganizationTokenCreateOptions{}
+
+	// Check whether the optional expiry was provided.
+	_, expiredAtProvided := d.GetOk("expired_at")
+
+	// If an expiry was provided, parse it and update the options struct.
+	if expiredAtProvided {
+		expiredAt := d.Get("expired_at").(string)
+
+		expiry, err := time.Parse(time.RFC3339, expiredAt)
+
+		options.ExpiredAt = &expiry
+
+		if err != nil {
```

Comment:
Could we please add a test to check that we are failing correctly when the string provided as the expiry is not in an acceptable format (for both team and org tokens)?

Reply:
Added! ✅